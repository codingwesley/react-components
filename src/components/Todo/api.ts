const fetcher = (url: string) => {
  return fetch(url).then((res) => {
    return res.json();
  });
};

export function fetchTodoList() {
  return fetcher("/api/todos");
}
