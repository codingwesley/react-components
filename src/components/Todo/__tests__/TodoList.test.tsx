import { render } from "react-dom";
import { act } from "react-dom/test-utils";
import { Todo } from "..";
import * as api from "../api";

const todos = [
  { id: "1", name: "wesley.chen", complete: true },
  { id: "2", name: "joy", complete: false },
];

const originalError = console.error;
beforeAll(() => {
  console.error = (...args) => {
    if (/Warning.*not wrapped in act/.test(args[0])) {
      return;
    }
    originalError.call(console, ...args);
  };
});

afterAll(() => {
  console.error = originalError;
});

describe("useTodoList test", () => {
  let mockFetchTodoList = jest.spyOn(api, "fetchTodoList");
  let container: null | HTMLDivElement = null;

  beforeEach(() => {
    mockFetchTodoList.mockResolvedValue(todos);
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    jest.clearAllMocks();
    if (container) {
      document.body.removeChild(container);
      container = null;
    }
  });

  it("fetch api call", () => {
    act(() => {
      render(<Todo title="test title" />, container);
    });

    expect(mockFetchTodoList).toBeCalled();
    expect(container?.textContent).toContain("test title")
  });
});
