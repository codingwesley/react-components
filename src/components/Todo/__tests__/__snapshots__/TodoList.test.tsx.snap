// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTodoList test item toggle 1`] = `
<div>
  <main
    class="ant-layout-content"
    style="margin: 10px 12px;"
  >
    <h2>
      test title
    </h2>
    <div
      class="ant-alert ant-alert-success ant-alert-no-icon"
      data-show="true"
      role="alert"
    >
      <div
        class="ant-alert-content"
      >
        <div
          class="ant-alert-message"
        >
          Completed count: 2
        </div>
      </div>
    </div>
    <div
      class="ant-row"
      data-role="todo-item"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <label
            class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
          >
            <span
              class="ant-checkbox ant-checkbox-checked"
            >
              <input
                checked=""
                class="ant-checkbox-input"
                id="todo-item-1"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              <span>
                wesley.chen
              </span>
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      class="ant-row"
      data-role="todo-item"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <label
            class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
          >
            <span
              class="ant-checkbox ant-checkbox-checked"
            >
              <input
                class="ant-checkbox-input"
                id="todo-item-2"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              <span>
                joy
              </span>
            </span>
          </label>
        </div>
      </div>
    </div>
  </main>
</div>
`;
